### PR TITLE
0.64-RC0 merged

### PR DESCRIPTION
**Pull request by HeyImChris**
#### Please select one of the following
- [X] I am removing an existing difference between facebook/react-native and microsoft/react-native-macos :thumbsup:
- [ ] I am cherry-picking a change from Facebook's react-native into microsoft/react-native-macos :thumbsup:
- [X] I am making a fix / change for the macOS implementation of react-native
- [X] I am making a change required for Microsoft usage of react-native

## Summary

I git cherry-picked a [commit](https://github.com/facebook/react-native/commit/69ce9c21d433a23ffb9934062b46fa64277ee255) from "close" to when we last forked. There were more changes than expected but in an attempt to update our fork in a more iterative/controlled process, this is the first batch of new changes/fixes that can come in to master.

I plan to repeat this process for changes in batches of a month's worth or so, so if something breaks, it's easier to trace back to when it happened.

## Changelog

[macOS/iOS] [Feature] - Merge several hundred commits into our fork

## Test Plan

Test apps run. Clicked through a half dozen or so controls on both platforms and all seems to be working.

https://user-images.githubusercontent.com/48299693/111264697-3ea5f280-85e5-11eb-9867-797e3761f398.mov

### CODE REVIEW

**Review by Saadnajmi — APPROVED** [review]

File: Libraries/Components/CheckBox/AndroidCheckBoxNativeComponent.js [hunk 1]
```diff
@@ -1,74 +0,0 @@
-/**
```
Comment (NickGerleman) [inline Libraries/Components/CheckBox/AndroidCheckBoxNativeComponent.js:1]: Seems like we can remove https://github.com/microsoft/react-native-macos/blob/master/Libraries/Components/CheckBox/CheckBox.macos.js

Comment (NickGerleman) [reply Libraries/Components/CheckBox/AndroidCheckBoxNativeComponent.js:1]: And any Mac specific VM code for it if there was any

**Comment by NickGerleman** [comment]
> I git cherry-picked a commit from "close" to when we last forked. There were more changes than expected but in an attempt to update our fork in a more iterative/controlled process, this is the first batch of new changes/fixes that can come in to master.

This strategy makes a lot of sense to me 👍.

A bit curious about where "RC0" comes from though. This gets us to May 2020, but the upstream RC 0 was for November 2020. IIRC Eloy did some work to publish master builds using different versioning?

**Comment by HeyImChris** [comment]
> > I git cherry-picked a commit from "close" to when we last forked. There were more changes than expected but in an attempt to update our fork in a more iterative/controlled process, this is the first batch of new changes/fixes that can come in to master.
> 
> This strategy makes a lot of sense to me 👍.
> 
> A bit curious about where "RC0" comes from though. This gets us to May 2020, but the upstream 0.64.0-rc.0 was based on a commit from November 2020. IIRC Eloy did some work to publish master builds using different versioning? Something like "0.0.0-hash" I think.

Yea I think the fb tags threw me a bit here. This commit is tagged with the `v0.64-rc.0` tag so it is one of the changes in RC0 but with the split happening a few months later I think it's just that. I'm still looking into 

> > I git cherry-picked a commit from "close" to when we last forked. There were more changes than expected but in an attempt to update our fork in a more iterative/controlled process, this is the first batch of new changes/fixes that can come in to master.
> 
> This strategy makes a lot of sense to me 👍.
> 
> A bit curious about where "RC0" comes from though. This gets us to May 2020, but the upstream 0.64.0-rc.0 was based on a commit from November 2020. IIRC Eloy did some work to publish master builds using different versioning? Something like "0.0.0-hash" I think.

Yea it looks like the Facebook tagging threw me a bit here. This commit I referenced is tagged as `v0.64-rc.0` but with the actual release coming a few months later, this must have just been some of the early lead up to the release.

**Comment by NickGerleman** [comment]
@HeyImChris I think the view you might have been looking at is the commits that a tag shows up in. If you expand it, it will just list off that the commit is present in the RC, etc. See the bottom here. 

72294CA7-2D41-47F3-A7C0-7E8B5B809CC9

**Pull request closed without merging**